Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -522,26 +530,20 @@ else if (attribute.getAttributeType().getValueType() == BlackboardAttribute.TSK_
      * @return
      */
     private static Lookup createLookup(BlackboardArtifact artifact) {
-        List<Object> forLookup = new ArrayList<>();
-        forLookup.add(artifact);
-
         // Add the content the artifact is associated with
-        Content content = getAssociatedContent(artifact);
-        if (content != null) {
-            forLookup.add(content);
-        }
 
-        return Lookups.fixed(forLookup.toArray(new Object[forLookup.size()]));
-    }
-
-    private static Content getAssociatedContent(BlackboardArtifact artifact) {
         try {
-            return artifact.getSleuthkitCase().getContentById(artifact.getObjectID());
-        } catch (TskCoreException ex) {
-            LOGGER.log(Level.WARNING, "Getting file failed", ex); //NON-NLS
+            Content content = contentCache.get(artifact.getObjectID(), () -> artifact.getSleuthkitCase().getContentById(artifact.getObjectID()));
+            if (content == null) {
+                return Lookups.fixed(artifact);
+            } else {
+                return Lookups.fixed(artifact, content);
+            }
+        } catch (ExecutionException ex) {
+            //JMTODO: Clean this up.  Either swallow it with appropriate logging/notification, or propogate it, not both.
```

Comment:
Let's resolve this TODO, please. I would not throw an unchecked exception here, and there is nothing for the calling code to do, so I'm inclined to  log the error and continue to construct the BlackboardArtifactNode with an incomplete lookup. The consequence would be evident ion the content viewers, right? Thoughts?